bionic/deriver.py

```diff
@@ -219,6 +216,10 @@ def _populate_entity_info(self, entity_name):
             dep_key_spaces_by_name,
             dep_task_key_lists_by_name)
 
+        ds = provider.attrs.docs
+        if ds[0]:
```

Comment:
I know you have a `TODO` to replace this code, so maybe you already know this, but just in case: the first element of `docs` will not necessarily be the doc corresponding to `entity_name`. That's why you want to use `provider.doc_for_name(entity_name)` to select the correct one.